src/operator-docs/combination/forkJoin.ts

```diff
@@ -1,6 +1,139 @@
+        Observables completes without emitting any value,
+        <span class="markdown-code">forkJoin</span> will complete at that moment as well and it will not
+        emit anything either, even if it already has some last values from other Observables.
+      </p>
+
+      <p>
+        Conversely, if there is an Observable that never completes,
+        <span class="markdown-code">forkJoin</span> will never complete as well, unless at
+        any point some other Observable completes without emitting value, which brings us back to the previous case.
+        Overall, in order for <span class="markdown-code">forkJoin</span> to emit a value, all Observables passed as arguments
+        have to emit something at least once and complete.
+      </p>
+      <p>
+        If any input Observable errors at some point, <span class="markdown-code">forkJoin</span>
+        will error as well and all other Observables
+        will be immediately unsubscribed.
+      </p>
+      <p>
+        Optionally <span class="markdown-code">forkJoin</span> accepts project function, that will be
+        called with values which normally would land in emitted array. Whatever is returned by project function,
+        will appear in output Observable instead. This means that default project can be thought of as a
+        function that takes all its arguments and puts them into an array. Note that project function will be called only
+        when output Observable is supposed to emit a result.
+      </p>
+    `
+  },
+  examples: [
+    {
+      name: 'Use forkJoin with operator emitting immediately',
+      code: `
+      const observable = Rx.Observable.forkJoin(
+        Rx.Observable.of(1, 2, 3, 4),
+        Rx.Observable.of(5, 6, 7, 8)
+      );
+      observable.subscribe(
+        value => console.log(value),
+        err => {},
+        () => console.log('This is how it ends!')
+      );
+      // Logs:
+      // [4, 8]
+      // "This is how it ends!"
+    `,
+      externalLink: {
+        platform: 'JSBin',
+        url: 'http://jsbin.com/kinilaruki/1/embed?js,console'
+      }
+    },
+    {
+      name: 'Use forkJoin with operator emitting after some time',
+      code: `
+      const observable = Rx.Observable.forkJoin(
+        Rx.Observable.interval(1000).take(3), // emit 0, 1, 2 every second and complete
+        Rx.Observable.interval(500).take(4) // emit 0, 1, 2, 3 every half a second and complete
+      );
+      observable.subscribe(
+        value => console.log(value),
+        err => {},
+        () => console.log('This is how it ends!')
+      );
+      // Logs:
+      // [2, 3] after 3 seconds
+      // "This is how it ends!" immediately after
+    `,
+      externalLink: {
+        platform: 'JSBin',
+        url: 'http://jsbin.com/rewivubuqi/1/embed?js,console'
+      }
+    },
+    {
+      name: 'Use forkJoin with project function',
+      code: `
+      const observable = Rx.Observable.forkJoin(
```

Comment:
Let's use es6 imports - I'll stop commenting on this knowing that we should update a few of these PRs. :) // @btroncone @ashwin-sureshkumar @sumitarora

Reply:
Other than that looks great!